src/com/google/javascript/jscomp/JSModuleGraph.java

```diff
@@ -452,26 +452,30 @@ public void coalesceDuplicateFiles() {
         entryPointInputs.addAll(sorter.getInputsWithoutProvides());
       }
 
-      for (String entryPoint : depOptions.getEntryPoints()) {
+      for (DependencyOptions.ModuleIdentifier entryPoint : depOptions.getEntryPoints()) {
         // An entry point is either formatted as:
         // 'foo.bar' - peg foo.bar to its current module
         // 'modC:foo.bar' - peg foo.bar to modC
-        String inputName = entryPoint;
-        int splitPoint = entryPoint.indexOf(':');
+        String inputName = entryPoint.getNormalizedModuleName();
+        int splitPoint = entryPoint.getNormalizedModuleName().indexOf(':');
         CompilerInput entryPointInput = null;
-        if (splitPoint != -1) {
-          String moduleName = entryPoint.substring(0, splitPoint);
-          inputName = entryPoint.substring(
-              Math.min(splitPoint + 1, entryPoint.length() - 1));
-          JSModule module = modulesByName.get(moduleName);
-          if (module == null) {
-            throw new MissingModuleException(moduleName);
+        try {
+          if (splitPoint != -1) {
+            String moduleName = entryPoint.getNormalizedModuleName().substring(0, splitPoint);
+            inputName = entryPoint.getNormalizedModuleName().substring(
+                Math.min(splitPoint + 1, entryPoint.getNormalizedModuleName().length() - 1));
+            JSModule module = modulesByName.get(moduleName);
+            if (module == null) {
+              throw new MissingModuleException(moduleName);
+            } else {
+              entryPointInput = sorter.getInputProviding(inputName);
+              entryPointInput.overrideModule(module);
+            }
           } else {
             entryPointInput = sorter.getInputProviding(inputName);
-            entryPointInput.overrideModule(module);
           }
-        } else {
-          entryPointInput = sorter.getInputProviding(inputName);
+        } catch (MissingProvideException e) {
+          throw new MissingProvideException(entryPoint.getName());
```

Comment:
please make sure to include the original exception (and stack trace!) as the cause, it makes debugging much easier.